indispensable-cosef/src/commonMain/kotlin/at/asitplus/signum/indispensable/cosef/CoseAlgorithm.kt

```diff
@@ -18,7 +17,7 @@ import kotlinx.serialization.encoding.Encoder
  * See [COSE Algorithm Registry](https://www.iana.org/assignments/cose/cose.xhtml)
  */
 @Serializable(with = CoseAlgorithmSerializer::class)
-enum class CoseAlgorithm(val value: Int): SpecializedSignatureAlgorithm {
+enum class CoseAlgorithm(val value: Int) : SpecializedDataIntegrityAlgorithm {
```

Comment:
are we sure we want to keep this as an `enum class` instead of having a proper class hierarchy (with `CoseAlgorithm` -> `CoseSignatureAlgorithm`/`CoseMAC` etc) so that we can have proper `Specialized*` conversions?

Reply:
delegating to #249